.drone.yml

```diff
@@ -1,23 +1,36 @@
 matrix:
   NAME:
     - ukhomeofficedigital/cdp-ci
-  DOCKER_USERNAME:
-    - ukhomeofficedigital+cdp_ci
   DOCKER_REPO:
     - quay.io
 pipeline:
   docker-build:
-    image: ukhomeoffice/drone-docker
-    repo: '${DOCKER_REPO}/${NAME}'
-    dry_run: true
+    image: docker
+    secrets:
+      - docker_username
+      - git_deployment_key
+    environment:
+      - DOCKER_HOST=tcp://172.17.0.1:2375
+    commands:
+      - docker build --build-arg "GIT_DEPLOYMENT_KEY=$${GIT_DEPLOYMENT_KEY}" -t ${DOCKER_REPO}/${NAME}:$${DRONE_COMMIT_SHA} .
     when:
-      event: pull_request
+      event:
+        # - pull_request
```

Comment:
Should this line be removed or uncommented?

Reply:
I'd think it should be uncommented.

Reply:
ha, we found a bug with that; for whatever reason, the pull_request option is not working as expected; we found that it often did not pick up the latest commit.

Reply:
I was thinking about this yesterday evening and I want to try re-creating the secret. I'm wondering if enabling Drone to act on PRs was set up after the secret was created, in which case that would be the reason why the secret has always been empty when the pipeline tried using it.

Reply:
There is definitely a bug around this. It seems to happen when you force push to a branch. It's quite annoying.

Apparently it's due to some caching on GitHub's end. You can usually work around it by doing in to drone a repeating the build.

If the idea of leaving it present but commented is to point out that we would prefer to do the build were the bug to not exist I think that is legitimate. Though we should probably add a secondary comment to give some context.

It's debatable whether we should avoid it entirely by not doing PR builds. If we require branches to be up to date with master before we merge then I suppose we shouldn't need PR builds anyway. (As the push build would be identical.)

@chrisns: Thoughts?

Reply:
Yeah, we've definitely had issues after an interactive rebase with fixups followed by a force push. In that case, I had to push an empty commit. But even then it wasn't entirely reliable / repeatable: when the pipeline build was triggered by a push, it was successful; yet it failed when triggered by a PR althought the pipeline steps were identical

Reply:
I think eventually GitHub updates its cache so it should just be a case of waiting and rebuilding.

These links seem to be related:
- https://github.com/drone-plugins/drone-git/issues/41#issuecomment-345746332
- https://discourse.drone.io/t/github-claims-that-merge-refs-are-undocumented-feature/1100

Reply:
Yes, if we enable the status check `Require branches to be up to date before merging` which seems sane then I agree PR builds and their events are redundant in terms of the pipeline

Reply:
re-enable the 'pull_request' event in the Drone pipeline after re-creating drone secret GIT_DEPLOYMENT_KEY. Still a failure in the pipeline when triggered from the pull request. Yet when triggered from a push, it's successful. The pipeline step between pull_request and push is shared...
I suggest we take the pull_request event out again - by commenting it out and adding a comment describing why that's the case

Reply:
That sounds reasonable to me.